walletlibrary/src/main/java/com/microsoft/walletlibrary/ExtensionIdentifierManager.kt

```diff
@@ -0,0 +1,78 @@
+
+package com.microsoft.walletlibrary
+
+import com.microsoft.walletlibrary.ExtensionIdentifierManager.Constants.SELF_ISSUED_ISSUER_NAME
+import com.microsoft.walletlibrary.did.sdk.credential.models.VerifiableCredentialContent
+import com.microsoft.walletlibrary.did.sdk.credential.models.VerifiableCredentialDescriptor
+import com.microsoft.walletlibrary.did.sdk.credential.service.protectors.createIssuedAndExpiryTime
+import com.microsoft.walletlibrary.did.sdk.util.controlflow.Result
+import com.microsoft.walletlibrary.did.sdk.util.log.SdkLog
+import com.microsoft.walletlibrary.networking.entities.openid4vci.credentialmetadata.CredentialConfiguration
+import com.microsoft.walletlibrary.util.LibraryConfiguration
+import com.microsoft.walletlibrary.verifiedid.OpenId4VciVerifiedId
+import com.microsoft.walletlibrary.verifiedid.VerifiedId
+import kotlinx.coroutines.runBlocking
+import java.util.UUID
+
+class ExtensionIdentifierManager internal constructor(libraryConfiguration: LibraryConfiguration) {
+    private val identifierManager = libraryConfiguration.identifierManager
+    private val serializer = libraryConfiguration.serializer
+    private val signer = libraryConfiguration.tokenSigner
+
+    internal object Constants {
+        const val VC_DATA_MODEL_CONTEXT = "https://www.w3.org/2018/credentials/v1"
+        const val VC_DATA_MODEL_TYPE = "VerifiableCredential"
+        const val SELF_ISSUED_ISSUER_NAME = "Self"
+    }
+
+    fun createEphemeralSelfSignedVerifiedId(
+        claims: Map<String, String>,
+        types: Array<String>
+    ): VerifiedId? {
+        try {
+            val vcTypes = mutableListOf(Constants.VC_DATA_MODEL_TYPE)
+            vcTypes += types
+            val vcDescriptor = VerifiableCredentialDescriptor(
+                listOf(Constants.VC_DATA_MODEL_CONTEXT), vcTypes, claims
+            )
+            val identifier = runBlocking {
+                when (val result =
+                    this@ExtensionIdentifierManager.identifierManager.getMasterIdentifier()) {
+                    is Result.Success -> result.payload
+                    is Result.Failure -> {
+                        SdkLog.e("Could not get DID", result.payload)
+                        null
+                    }
+                }
+            } ?: return null
+
+            val (issuedTime, expiryTime) = createIssuedAndExpiryTime(5 * 60)    // 5 minutes
+            val jti = UUID.randomUUID().toString()
+            val content = VerifiableCredentialContent(
+                jti,
+                vcDescriptor,
+                identifier.id,
+                identifier.id,
+                issuedTime,
+                expiryTime
+            )
+            val jsonContent = serializer.encodeToString(VerifiableCredentialContent.serializer(), content)
+            val vcToken = signer.signWithIdentifier(jsonContent, identifier)
+            return OpenId4VciVerifiedId(
```

Comment:
This returns a OpenId4VciVerifiedId. We are using it only for Issuance using OpenId4Vci protocol. There are instances where raw is extracted from specific type of Verified Id, those paths would fail. We will have to change those to use a serializer to get the raw instead of directly getting it.

Reply:
Are those paths along scenarios that pull verified IDs from the database? This function is explicitly for creating short-lived VerifiedIDs for the purpose of single-use self-signed responses.

Reply:
Raw is extracted from Verified ID when issuance/presentation response is mapped and also in PresentationSubmissionGroup. We can make it consistent and use the VerifiedID serializer to have the logic in one place and not worry about checking for all the possibilities in all the paths.

Reply:
After pulling in your commit for credential serializer + calling that function in these locations, it should be fine to use this.